lib/jekyll/static_file.rb

```diff
@@ -144,7 +144,11 @@ def copy_file(dest_path)
       if @site.safe || Jekyll.env == "production"
         FileUtils.cp(path, dest_path)
       else
-        FileUtils.copy_entry(path, dest_path)
+        begin
+          FileUtils.ln(path, dest_path)
+        rescue
```

Comment:
Begin and Rescue should not to be used as flow control.  This is a blocker as this is highly disoptimized.